R/change_catch.R

```diff
@@ -0,0 +1,15 @@
+#' Change catch in the datafile.
+#'
+#' @param dat_list A data file as read in using `r4ss::SS_readdat()`
+#' @param f_params A list of parameters related to fishing mortality
+#' @return a modified data file as a list in R.
+#' @author Kathryn Doering
+#'
+change_catch <- function(dat_list, f_params) {
+  # Note: would need to modify if did not want catch every year, potentially.
+  dat_list[["catch"]] <- expand.grid(
+    year = c(-999, dat_list$styr:dat_list$endyr),
+    seas = 1, fleet = f_params$fleets, catch = 1, catch_se = 0.01)
```

Comment:
@k-doering-NOAA should we think about season being hard-coded here? I am mainly thinking in terms of what you found with Henry's model.

Reply:
good point - should we pull season from the dummy data file instead (and maybe if it is missing, just put the season as 1 with a warning)?

Reply:
i think it would be better to have it as part of f_params where the default is one

Reply:
that's a good idea

Reply:
Should we just change line 12 to
```
seas = seq(dat_list[["nseas"]]), fleet = f_params$fleets, catch = 1, catch_se = 0.01)
```

Reply:
I think a slightly more complicated solution was necessary - see d7c721f . I think we will need to do more thorough testing with seasonal models in the future, but think this PR is ready to be merged.